spring/src/main/java/org/axonframework/spring/config/SpringAxonAutoConfigurer.java

```diff
@@ -424,18 +423,34 @@ public void setBeanFactory(BeanFactory beanFactory) throws BeansException {
 
         @Override
         public void initialize(Configuration config) {
-            delegate = delegateSupplier.get();
-            delegate.initialize(config);
+            getDelegate().initialize(config);
         }
 
         @Override
         public void start() {
-            delegate.start();
+            getDelegate().start();
         }
 
         @Override
         public void shutdown() {
-            delegate.shutdown();
+            getDelegate().shutdown();
+        }
+
+        @Override
+        public int phase() {
+            return getDelegate().phase();
+        }
+
+        @Override
+        public ModuleConfiguration instance() {
+            return getDelegate();
+        }
+
+        private ModuleConfiguration getDelegate() {
+            if (delegate == null) {
```

Comment:
Nit: You could change this to a one liner: `return delegate == null ? delegateSupplier.get() : delegate;`

Reply:
This way delegate will never be cached.

Reply:
With 'this way', do you mean the use of the tertiare operator or the if-statement?

Reply:
With ternary operator I mean, since `delegate` will never get the value from the `delegateSupplier`.

Reply:
ah yes of course >_>